### PR TITLE
Add robot name

### PR DESCRIPTION
**Pull request by stevejb71**
I think this is a useful exercise to add, as it _requires_ mutable state for its solution. No other exercise I've seen does this.

This could be made an easier/less challenging exercise by removing the enforcing of unique names (the final test). It would still let learners find out about mutable state & randomness. 

What do you think?

### CODE REVIEW

**Comment by dvberkel** [comment]
Exploring the specific strengths and weaknesses of a language is very interesting. So I agree that it would be interesting to see how this pans out

File: exercises/robot-name/example.ml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+open Core.Std
+
+type robot = {mutable index : int}
+
+let index = ref (-1)
+
+let unique_ids: int array =
```
Comment (dvberkel) [inline exercises/robot-name/example.ml]: I really like how you prevent naming collisions

**Comment by stevejb71** [comment]
I rolled back a bit from requiring all unique names for the Robots. It's a little peripheral from what I think this exercise can teach (reference cells and mutable record fields), and makes the exercise significantly harder.

However I think it's worth keeping as an option if anyone wants a bit more of a challenge, so I left a test in which is by default not enabled (I really wish ounit had the idea of a pending or ignored test!)

**Comment by stevejb71** [comment]
I'll remove the dependency on the re2 regex library. It's just being used in the tests. Will be a shame to force this dependency on the student, as they will almost certainly not want it for the implementation.

**Comment by dvberkel** [comment]
Rolling back uniqueness is probably better, because it could burden people a lot.

**Comment by stevejb71** [comment]
Removed the re2 dependency, so I think is good to go.

**Pull request merged**